Implement utility activate load info guijob

herve-er opened the pull request.

herve-er committed:
feat: add ActivateLoadInfo gui job

herve-er committed:
feat(test): add UTILITY_ACTIVATELOADINFO job test

herve-er committed:
chore: Fix typo and unused variable.

herve-er committed:
fix(test): update expected JSON with "type" in params

luc-guyot-infomaniak commented on src/server/CMakeLists.txt:
```diff
@@ -83,6 +83,9 @@ set(server_SRCS
     comm/guijobs/nodecreatemissingfoldersjob.h comm/guijobs/nodecreatemissingfoldersjob.cpp
     comm/guijobs/parametersinfojob.h comm/guijobs/parametersinfojob.cpp
     comm/guijobs/parametersupdatejob.h comm/guijobs/parametersupdatejob.cpp
+    comm/guijobs/blacklistednodelistjob.h comm/guijobs/blacklistednodelistjob.cpp
```
These two are already included above.

herve-er replied:
Done here: https://github.com/Infomaniak/desktop-kDrive/pull/1282/commits/13092aa0911c986560e60299267730c0357290b4

luc-guyot-infomaniak commented on src/server/comm/guijobs/activateloadinfojob.cpp:
```diff
@@ -0,0 +1,49 @@
+
+#include "activateloadinfojob.h"
+#include "appserver.h"
+#include "requests/serverrequests.h"
```
You probably don't need this one and the next.

herve-er replied:
Done here: https://github.com/Infomaniak/desktop-kDrive/pull/1282/commits/13092aa0911c986560e60299267730c0357290b4

luc-guyot-infomaniak commented on src/server/comm/guijobs/activateloadinfojob.cpp:
```diff
@@ -0,0 +1,49 @@
+}
+
+ExitInfo ActivateLoadInfoJob::process() {
+    // Clear sync cache and reload user information
```
The names of the two methods below are sufficiently suggestive, I believe. We can omit this comment.

herve-er replied:
Done here: https://github.com/Infomaniak/desktop-kDrive/pull/1282/commits/13092aa0911c986560e60299267730c0357290b4

luc-guyot-infomaniak commented on src/server/comm/guijobs/guijobfactory.cpp:
```diff
@@ -83,7 +84,9 @@ GuiJobFactory::GuiJobFactory() {
-                {RequestNum::PARAMETERS_UPDATE, makeShared<ParametersUpdateJob>}};
+                {RequestNum::PARAMETERS_UPDATE, makeShared<ParametersUpdateJob>},
+                {RequestNum::UTILITY_ACTIVATELOADINFO, makeShared<ActivateLoadInfoJob>},
+                {RequestNum::NODE_CREATEMISSINGFOLDERS, makeShared<NodeCreateMissingFoldersJob>}};
```
Already added above.

herve-er replied:
Done here: https://github.com/Infomaniak/desktop-kDrive/pull/1282/commits/13092aa0911c986560e60299267730c0357290b4

luc-guyot-infomaniak commented on test/server/CMakeLists.txt:
```diff
@@ -61,6 +61,9 @@ set(server_SRCS
     ${server_srcs_path}/comm/guijobs/nodesubfolders2job.h ${server_srcs_path}/comm/guijobs/nodesubfolders2job.cpp
     ${server_srcs_path}/comm/guijobs/nodefoldersizejob.h ${server_srcs_path}/comm/guijobs/nodefoldersizejob.cpp
     ${server_srcs_path}/comm/guijobs/nodecreatemissingfoldersjob.h ${server_srcs_path}/comm/guijobs/nodecreatemissingfoldersjob.cpp
+    ${server_srcs_path}/comm/guijobs/blacklistednodelistjob.h ${server_srcs_path}/comm/guijobs/blacklistednodelistjob.cpp
```
These two have already been include above.

herve-er replied:
Done here: https://github.com/Infomaniak/desktop-kDrive/pull/1282/commits/13092aa0911c986560e60299267730c0357290b4

luc-guyot-infomaniak commented on test/server/comm/guicommchannel/testguicommchannel.h:
```diff
@@ -81,6 +81,7 @@ class TestGuiCommChannel : public CppUnit::TestFixture, public TestBase {
          */
         CPPUNIT_TEST(testNodeSubFolders2Job);
         CPPUNIT_TEST(testNodeCreateMissingFoldersJob);
+        CPPUNIT_TEST(testActivateLoadInfoJob);
```
If we follow the order prescribed by `comm.h`,  the test and the method `testActivateLoadInfoJob` should be at the bottom.

herve-er replied:
Done here: https://github.com/Infomaniak/desktop-kDrive/pull/1282/commits/13092aa0911c986560e60299267730c0357290b4

luc-guyot-infomaniak commented on test/server/comm/guicommchannel/testnodejobs.cpp:
```diff
@@ -275,4 +275,40 @@ void TestGuiCommChannel::testNodeCreateMissingFoldersJob() {
+                         R"( "params": { }, "type": )" +
+                         std::to_string(toInt(AbstractGuiJob::GuiJobType::Query)) + R"( })"};
+
+    auto processFct = [](std::shared_ptr<AbstractGuiJob> job) {};
```
You may check that the job has the expected type.

herve-er replied:
Done here: https://github.com/Infomaniak/desktop-kDrive/pull/1282/commits/13092aa0911c986560e60299267730c0357290b4

luc-guyot-infomaniak commented on test/server/comm/guicommchannel/testnodejobs.cpp:
```diff
@@ -275,4 +275,40 @@ void TestGuiCommChannel::testNodeCreateMissingFoldersJob() {
 #endif
 }
 
+void TestGuiCommChannel::testActivateLoadInfoJob() {
```
Why not creating a new file named `testutility.cpp` where the unit tests referring to `UTILITY_*` jobs would be implemented? Why not using the full enum name in the name of the test, i.e.,  `testUtilityActivateLoadInfoJob`?

herve-er replied:
Done here: https://github.com/Infomaniak/desktop-kDrive/pull/1282/commits/13092aa0911c986560e60299267730c0357290b4

luc-guyot-infomaniak commented on src/server/comm/guijobs/utilityactivateloadinfojob.h:
```diff
@@ -0,0 +1,38 @@
+/*
```
About the name of the job: why omitting the `utility` prefix?

herve-er replied:
Done here: https://github.com/Infomaniak/desktop-kDrive/pull/1282/commits/13092aa0911c986560e60299267730c0357290b4

herve-er committed:
chore: Adress PR comments

herve-er committed:
refactor(testnodejobs): remove unused header include

herve-er committed:
fix(tests): remove cbkAnswer assertion, adjust JSON format

Removed platform-specific cbkAnswer assertions in testGenericJob.
Adjusted JSON formatting in testActivateLoadInfoJob by adding
an extra space in the "params" object. These changes may affect
test behavior and string comparisons.

herve-er committed:
chore: Minor PR fix

herve-er committed:
fix: add missing ERROR_INFOLIST job support

herve-er committed:
chore: Minor syntax fix and removal of a duplicate include

herve-er committed:
Merge branch 'develop' into implement-utility-activate-load-info-guijob

herve-er committed:
refactor(test): rename testActivateLoadInfoJob method

Renamed testActivateLoadInfoJob to testUtilityActivateLoadInfoJob in TestGuiCommChannel, updating both the test suite declaration and the test implementation. No functional changes introduced.

herve-er committed:
fix(test): add testUtilityActivateLoadInfoJob declaration
